webrender/src/util.rs

```diff
@@ -209,3 +209,116 @@ pub fn subtract_rect(rect: &Rect<f32>,
         }
     }
 }
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[repr(u8)]
+pub enum TransformedRectKind {
+    AxisAligned = 0,
+    Complex = 1,
+}
+
+#[derive(Debug, Clone)]
+pub struct TransformedRect {
+    pub local_rect: Rect<f32>,
+    pub bounding_rect: Rect<DevicePixel>,
+    pub vertices: [Point4D<f32>; 4],
+    pub kind: TransformedRectKind,
+}
+
+impl TransformedRect {
+    pub fn new(rect: &Rect<f32>,
+           transform: &Matrix4D<f32>,
+           device_pixel_ratio: f32) -> TransformedRect {
+
+        let kind = if transform.can_losslessly_transform_and_perspective_project_a_2d_rect() {
+            TransformedRectKind::AxisAligned
+        } else {
+            TransformedRectKind::Complex
+        };
+
+/*
```

Comment:
I prefer not to have commented out code, but I understand why you did it here. I think it'd be good to add a `FIXME` at the top saying what the commented out code does and when to reenable it.